### DISCUSSION TITLE
Version 2.1.6

### DISCUSSION THREAD

**Discussion by rainepretorius** [rainepretorius/olarm-ha-integration#57] [Announcements]
**Bugfixes:**

- Working 100% again after implementing a workaround for the API limit.
- Tried to fix device selection not working.

**Improvements:**
Workaround for Olarm's request rate limiting that definitely works as the setup calls have been removed and replaced with just one call  to the new API endpoint.

<hr /><em>This discussion was created from the release <a href='https://github.com/rainepretorius/olarm-ha-integration/releases/tag/version2.1.6'>Version 2.1.6</a>.</em>

**Comment by MrPrisoner** [comment]
@rainepretorius works perfectly with my 1 device, thanks!

I've noticed when toggling a bypass switch it pops up the following error:
`Failed to call service switch/turn_off. object NoneType can't be used in 'await' expression`

Can be fixed by removing these awaits: https://github.com/MrPrisoner/olarm-ha-integration/commit/9238222c14aa1828a1105627bd183ce3d86627e4
